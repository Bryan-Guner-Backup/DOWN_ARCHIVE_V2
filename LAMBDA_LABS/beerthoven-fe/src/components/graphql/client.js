import { ApolloClient } from "apollo-client";
import { InMemoryCache } from "apollo-cache-inmemory";
import { ApolloLink } from "apollo-link";
import { HttpLink } from "apollo-link-http";
import { setContext } from "apollo-link-context";
import { RetryLink } from "apollo-link-retry";

// create retry link -- since it takes some time to get
// the Okta token, this retry will cause any failed Apollo
// requests to retry
const retryLink = new RetryLink();

// create authenticated link
const authLink = setContext((_, { headers }) => {
  // return the headers to the context
  const token = localStorage.getItem("okta-token");
  return {
    headers: {
      ...headers,
      Authorization: token ? `Bearer ${token}` : "",
    },
  };
});

// apollo client setup
const link = new HttpLink({
  uri: "https://apollo.beerthoven.dev",
});

// create the client
const client = new ApolloClient({
  cache: new InMemoryCache(),
  link: ApolloLink.from([retryLink, authLink, link]),
});

export default client;
