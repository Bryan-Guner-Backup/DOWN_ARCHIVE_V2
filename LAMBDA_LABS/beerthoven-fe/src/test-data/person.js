const person = {
	first_name : 'Bob',
	email      : 'bob.smith@bobsmith.net',
	last_name  : 'Smith',
	phone      : '9999999999',
	address    : '12355 4th PL NW',
	address2   : '#512',
	city       : 'Lake Stevens',
	state      : 'Washington',
	zip        : '98258',
};

export default person;
