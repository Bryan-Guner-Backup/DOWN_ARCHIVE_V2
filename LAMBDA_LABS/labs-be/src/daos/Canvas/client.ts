import { Request, Response } from "express";
import {
  BaseClient,
  AuthTypes,
  ClientResponse,
  ClientArrayResponse,
} from "@shared/BaseClient";

export interface ICanvasClient<T> {
  get: (path: string) => ClientResponse<T> | ClientArrayResponse<T>;
}

class CanvasClient<T> extends BaseClient<T> implements ICanvasClient<T> {
  constructor() {
    const opts = {
      token: process.env.CANVAS_ACCESS_TOKEN,
      baseUrl: "https://lambdaschool.instructure.com/api/v1/",
      authType: AuthTypes.JWT,
    };

    super(opts);
  }
}

export default CanvasClient;
