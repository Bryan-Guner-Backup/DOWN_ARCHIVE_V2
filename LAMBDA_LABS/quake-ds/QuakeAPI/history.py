from flask import jsonify
from math import degrees, radians, cos, sin, asin, atan2, sqrt


def history(lat, lon, dist):
    '''Calculate the top left and bottom right coordinates at distance dist
    from coordinates lat, lon'''

    lat = radians(lat)
    lon = radians(lon)
    radius = 6371
    top_left = radians(315)
    bottom_right = radians(135)

    lat2A = asin(sin(lat) * cos(dist/radius) +
                 cos(lat) * sin(dist/radius) * cos(top_left))
    lat2B = asin(sin(lat) * cos(dist/radius) +
                 cos(lat) * sin(dist/radius) * cos(bottom_right))

    lon2A = lon + atan2(sin(top_left) * sin(dist/radius) *
                        cos(lat), cos(dist/radius) - sin(lat) * sin(lat2A))
    lon2B = lon + atan2(sin(bottom_right) * sin(dist/radius) *
                        cos(lat), cos(dist/radius) - sin(lat) * sin(lat2B))

    coordinates = {
        'lonA': degrees(lon2A),
        'latA': degrees(lat2A),
        'lonB': degrees(lon2B),
        'latB': degrees(lat2B)
    }


    return coordinates
