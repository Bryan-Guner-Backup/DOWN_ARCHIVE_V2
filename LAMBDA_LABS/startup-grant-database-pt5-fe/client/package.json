{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^4.9.12",
    "@material-ui/icons": "^4.9.1",
    "@material-ui/lab": "^4.0.0-alpha.51",
    "axios": "^0.19.2",
    "clsx": "^1.1.0",
    "firebase": "^7.14.2",
    "firebaseui": "^4.5.0",
    "material-ui": "^0.20.2",
    "material-ui-dropzone": "^2.5.0",
    "moment": "^2.24.0",
    "qs": "^6.9.3",
    "react": "^16.13.1",
    "react-datetime": "^2.16.3",
    "react-dom": "^16.13.1",
    "react-moment": "^0.9.7",
    "react-multiselect-checkboxes": "^0.1.1",
    "react-router-dom": "^5.1.2",
    "react-scripts": "^3.4.1",
    "use-global-hook": "^0.1.12"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "run:cypress": "$(npm bin)/cypress open"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "cypress": "^3.8.3"
  }
}
