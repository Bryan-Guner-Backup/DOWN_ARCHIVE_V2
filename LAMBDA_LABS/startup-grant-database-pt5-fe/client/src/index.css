body {
  margin: 0;
  font-family: 'Lato',
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}


@font-face {
  font-family: 'Lato';
  src: local('Lato'), url(./resources/fonts/Lato/Lato-Regular.ttf) format('truetype');
}