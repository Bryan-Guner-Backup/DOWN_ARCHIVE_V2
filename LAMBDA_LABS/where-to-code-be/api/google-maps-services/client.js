const Client = require("@googlemaps/google-maps-services-js").Client;

module.exports = {
  googleLocation,
  googlePlacePhoto
};

const client = new Client({});
const key = process.env.GCP_KEY;

// googleLocation takes a Google Place Id and returns a promise
function googleLocation(googleId) {
  return client.placeDetails({
    params: {
      place_id: googleId,
      fields: "name,formatted_address,formatted_phone_number,icon,photos",
      key
    },
    timeout: 1000
  });
}

// googlePlacePhoto takes a reference Id and returns a promise
function googlePlacePhoto(photoreference) {
  return client.placePhoto({
    params: {
      photoreference,
      maxwidth: 300,
      key
    },
    timeout: 1000
  });
}
