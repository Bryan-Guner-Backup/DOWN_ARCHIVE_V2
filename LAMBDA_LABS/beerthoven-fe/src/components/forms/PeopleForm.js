import React, { useState } from 'react';
import { useParams } from 'react-router-dom';
import { useQuery } from '@apollo/react-hooks';
import { GET_PERSON } from '../graphql/queries';
import { Form, Input, Button, Row, Col } from 'antd';
import { Link } from 'react-router-dom';


const PeopleForm = ({ onSubmit }) => {
	const params = useParams();

	// if there is no ID in the path, the user is creating a new person,
	// and we can skip fetching a person
	const editingPerson = params.id != null;

	const { loading, data } = useQuery(GET_PERSON, {
		skip      : !editingPerson, // skip grabbing a person if user is not editing, but rather adding new
		variables : { id: params.id },
	});

	const [
		submitted,
		setSubmitted,
	] = useState(false);

	const submitForm = values => {
		setSubmitted(true);
		onSubmit(values, params.id);
	};

	const [
		form,
	] = Form.useForm();

	if (!loading && data) {
		form.setFieldsValue(data.person);
	}

	return !submitted ? (
		<Form form={form} layout='vertical' onFinish={submitForm}  data-testid="form">
			<h1>Add New Profile</h1>
			<Row
				gutter={[
					16,
					16,
				]}>
				<Col span={12}>
					<Form.Item
						label='First Name'
						name='first_name'
						rules={[
							{ required: true },
						]}>
						<Input placeholder='Enter first name' />
					</Form.Item>
				</Col>
				<Col span={12}>
					<Form.Item
						label='Last Name'
						name='last_name'
						rules={[
							{ required: true },
						]}>
						<Input placeholder='Enter last name' />
					</Form.Item>
				</Col>
			</Row>

			<Row
				gutter={[
					16,
					16,
				]}>
				<Col span={12}>
					<Form.Item
						label='Email'
						name='email'
						rules={[
							{ required: true },
						]}>
						<Input type='email' placeholder='Enter email address' />
					</Form.Item>
				</Col>
				<Col span={12}>
					<Form.Item label='Phone number' name='phone'>
						<Input placeholder='Enter phone number' />
					</Form.Item>
				</Col>
			</Row>

			<Row
				gutter={[
					16,
					16,
				]}>
				<Col>
					<Form.Item label='Address' name='address'>
						<Input placeholder='Enter street address' />
					</Form.Item>
				</Col>
			</Row>

			<Row
				gutter={[
					16,
					16,
				]}>
				<Col span={12}>
					<Form.Item label='Address line 2 (optional)' name='address2'>
						<Input placeholder='Enter apartment, suite, etc' />
					</Form.Item>
				</Col>
				<Col span={12}>
					<Form.Item label='City' name='city'>
						<Input placeholder='Enter city' />
					</Form.Item>
				</Col>
			</Row>

			<Row
				gutter={[
					16,
					16,
				]}>
				<Col span={6}>
					<Form.Item label='State' name='state'>
						<Input placeholder='Enter state' />
					</Form.Item>
				</Col>
				<Col span={6}>
					<Form.Item
						label='Zip code'
						name='zip'
						rules={[
							{ required: true },
						]}>
						<Input placeholder='Enter zip code' />
					</Form.Item>
				</Col>
				<Col span={12} />
			</Row>

			<Form.Item>
				<Button type='primary' htmlType='submit'>
					Submit
				</Button>
			</Form.Item>
		</Form>
	) : (
		<div>
			<p>Submitted successfully</p>
			<Link to='/people'>
				<Button>Continue</Button>
			</Link>
		</div>
	);
};

export default PeopleForm;
