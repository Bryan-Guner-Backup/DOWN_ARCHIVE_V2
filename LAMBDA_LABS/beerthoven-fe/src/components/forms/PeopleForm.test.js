import React from 'react';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';
import PeopleForm from './PeopleForm';
import person from '../../test-data/person';
import '@testing-library/jest-dom/extend-expect'
import {Router} from 'react-router'
import { createMemoryHistory } from "history";
import { ApolloProvider } from 'react-apollo';
import client from '../graphql/client';


//to see if mock info is able to be retrieved 
const onSubmit = jest.fn()
    

// const checkIfErrors =(data)=>{
//     if(!data.name){
//        return false
//     }
//     else {
//         return true
//     }
// }

 //this submit is to validate inputs
//  const onSubmit2 = data =>{  
//   const isValid= checkIfErrors(data)
//   jest.fn()

// if(!isValid){
//   return null
// }else{
//   return data
// }
// }


  describe(`People Form Component`, ()=>{
    beforeEach(cleanup);

 it(`renders a component that uses withRouter`, async()=>{
const history = createMemoryHistory()
const route = '/some-route'
history.push(route)
const {container} = render(
  <ApolloProvider client={client}>
<Router history={history}><PeopleForm/></Router>
</ApolloProvider>
)

expect(container.firstChild).toMatchSnapshot();
 })

 it(`should fetch the required data`, async()=>{
const history = createMemoryHistory()
const {getByLabelText, getByText} = render(
<ApolloProvider client={client}>
<Router history={history}><PeopleForm  person={person}/></Router>
</ApolloProvider>
)
  const firstNameNode = await getByLabelText('First Name');
    const lastNameNode = getByLabelText('Last Name');
    const emailNode = getByLabelText('Email');
    const zipNode = getByLabelText('Zip code');
    const submitBtn = await getByText('Submit');


    await wait(() => {
    fireEvent.change(firstNameNode, { target: { value: person.first_name } });
    fireEvent.keyDown(firstNameNode, {key: "Random naming"})
    fireEvent.change(lastNameNode, { target: { value: person.last_name } });
    fireEvent.change(emailNode, { target: { value: person.email } });
    fireEvent.change(zipNode, { target: { value: person.zip } });
    });
    

  
  await onSubmit(person)
  expect(onSubmit).toHaveBeenCalledTimes(1);
  expect(onSubmit).toHaveBeenCalled();
  expect(onSubmit).toHaveBeenCalled()

})
  

it('Data matches mock data as expected', async()=>{

  expect(onSubmit).toHaveBeenCalledWith({
  first_name: 'Bob',
  email: 'bob.smith@bobsmith.net',
  last_name: 'Smith',
  phone: '9999999999',
  address: '12355 4th PL NW',
  address2: '#512',
  city: 'Lake Stevens',
  state: 'Washington',
  zip: '98258' });
})

it('similuates a button click', async()=>{

  const history = createMemoryHistory()
const {getByText} = render(
<ApolloProvider client={client}>
<Router history={history}><PeopleForm person={person}/></Router>
</ApolloProvider>
)

  const submitBtn =  getByText('Submit');
  await wait (()=>{
    fireEvent.click(submitBtn); //test is red when fireEvent.submit()
  })
})


})
