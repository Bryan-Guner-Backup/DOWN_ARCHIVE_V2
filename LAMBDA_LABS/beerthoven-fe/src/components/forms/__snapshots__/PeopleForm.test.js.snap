// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People Form Component renders a component that uses withRouter 1`] = `
<form
  class="ant-form ant-form-vertical"
  data-testid="form"
>
  <h1>
    Add New Profile
  </h1>
  <div
    class="ant-row"
    style="margin: -8px -8px 8px -8px;"
  >
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="first_name"
            title="First Name"
          >
            First Name
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="first_name"
                placeholder="Enter first name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="last_name"
            title="Last Name"
          >
            Last Name
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="last_name"
                placeholder="Enter last name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -8px -8px 8px -8px;"
  >
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="email"
            title="Email"
          >
            Email
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="email"
                placeholder="Enter email address"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="phone"
            title="Phone number"
          >
            Phone number
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="phone"
                placeholder="Enter phone number"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -8px -8px 8px -8px;"
  >
    <div
      class="ant-col"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="address"
            title="Address"
          >
            Address
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="address"
                placeholder="Enter street address"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -8px -8px 8px -8px;"
  >
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="address2"
            title="Address line 2 (optional)"
          >
            Address line 2 (optional)
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="address2"
                placeholder="Enter apartment, suite, etc"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="city"
            title="City"
          >
            City
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="city"
                placeholder="Enter city"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row"
    style="margin: -8px -8px 8px -8px;"
  >
    <div
      class="ant-col ant-col-6"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="state"
            title="State"
          >
            State
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="state"
                placeholder="Enter state"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-6"
      style="padding: 8px 8px 8px 8px;"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="zip"
            title="Zip code"
          >
            Zip code
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                class="ant-input"
                id="zip"
                placeholder="Enter zip code"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-12"
      style="padding: 8px 8px 8px 8px;"
    />
  </div>
  <div
    class="ant-row ant-form-item"
  >
    <div
      class="ant-col ant-form-item-control"
    >
      <div
        class="ant-form-item-control-input"
      >
        <div
          class="ant-form-item-control-input-content"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
